src/main/java/issame/achievements_core/mixin/CakeBlockMixin.java

```diff
@@ -0,0 +1,38 @@
+package issame.achievements_core.mixin;
+
+import btw.block.blocks.CakeBlock;
+import btw.client.fx.BTWEffectManager;
+import issame.achievements_core.event.EventDispatcher;
+import net.minecraft.src.*;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@Mixin(CakeBlock.class)
+public abstract class CakeBlockMixin extends BlockCake {
+    public CakeBlockMixin(int iBlockID ) {
+        super(iBlockID);
+    }
+
+    @Inject(method="onNeighborBlockChange",
```

Comment:
I think this should be changed to `@Inject(method = "onNeighborBlockChange(Lnet/minecraft/src/World;IIII)V", at = @At(value = "INVOKE", target = "Lnet/minecraft/src/World;playAuxSFX(IIIII)V"))`.
This will inject code whenever the `playAuxSFX` method is called so you don't need to check if it's being powered (and will presumably be more resilient to code changes)